src/trace_processor/importers/ftrace/generic_ftrace_tracker.cc

```diff
@@ -20,18 +20,116 @@
 #include "perfetto/protozero/proto_utils.h"
 #include "protos/perfetto/common/descriptor.pbzero.h"
 
+#include "src/trace_processor/importers/common/event_tracker.h"
+#include "src/trace_processor/importers/common/process_tracker.h"
+#include "src/trace_processor/importers/common/slice_tracker.h"
+#include "src/trace_processor/importers/common/track_tracker.h"
+#include "src/trace_processor/importers/common/tracks.h"
+#include "src/trace_processor/importers/common/tracks_common.h"
 #include "src/trace_processor/storage/stats.h"
+#include "src/trace_processor/storage/trace_storage.h"
 
 namespace perfetto::trace_processor {
-
 using protozero::proto_utils::ProtoSchemaType;
 
+namespace {
+
 // We do not expect tracepoints with over 32 fields. It's more likely that the
 // trace is corrupted. See also |kMaxFtraceEventFields| in ftrace_descriptors.h.
 static constexpr uint32_t kMaxAllowedFields = 32;
 
+static constexpr char kScopeFieldNamePrefix[] = "scope_";
+
+// Track blueprints for kernel track events.
+//
+// TODO(rsavitski): improve track naming in the UI for cpu & custom scoped
```

Comment:
Shouldn't we just do this now? I feel like it shouldn't be that much work and would add non-trivial benefits.